R/epichains.R

```diff
@@ -55,19 +55,21 @@ new_epichains <- function(tree_df,
 #' @author James M. Azam
 #' @export
 epichains <- function(tree_df,
-                           index_cases,
-                           statistic,
-                           offspring_dist,
-                           stat_max,
-                           track_pop) {
+                      index_cases,
+                      statistic,
+                      offspring_dist,
+                      track_pop,
+                      stat_max = Inf) {
   # Check that inputs are well specified
-  checkmate::assert_data_frame(tree_df)
-  checkmate::assert_integerish(index_cases, null.ok = TRUE)
-  checkmate::assert_character(statistic, null.ok = TRUE)
+  checkmate::assert_data_frame(tree_df, min.cols = 3, min.rows = index_cases)
+  checkmate::assert_count(index_cases, positive = TRUE)
+  checkmate::assert_choice(statistic, choices = c("size", "length"))
```

Comment:
I suggested this recently to {epiCo} too: my preference would be to check for a string, and use `match.arg()` and place the choices vector in the function call. That makes it easier for users to quickly grok what choices are available without reading the fn docs.

Reply:
I considered that as well. It is done this way in the main `simulate_*()` functions (which are user-facing). Since this is an internal function, I decided not to do that. I'll revert to your suggested style for the sake of others reading the internal code.